File: chp2/swap.py
# same swap function

def swap(x, y):
    tmp = x # also, tmp is a temporary variable for safely keeping `x` value
    x = y
    y = tmp
    print(f"Inside swap: x = {x}, y = {y}\n") 

if __name__ == '__main__':
  x = 21
  y = 17
  # swap called, x = 17, y = 21
  swap(x,y)
  # inside main part, x = 21, y = 17
  print(f"Inside main: x = {x}, y = {y}\n")

File: chp2/swap.c
#include <stdio.h>
#include <stdlib.h>
// swap function: swapping two integers values
void swap(int a, int b) { 
	int t = a; // t is a temporary variable 
	a = b;
	b = t;
	printf("swap: a = %d, b = %d\n", a, b);
}
//main fucntion
int main() {
	int a = 21;
	int b = 17;
	swap(a, b); // swap called, a = 17, b = 21
	printf("main: a = %d, b = %d\n", a, b); // inside main, a = 21, b = 17
	return EXIT_SUCCESS;
}
